library_generation/model/config_change.py

```diff
@@ -156,3 +171,31 @@ def __create_qualified_commit(
         if len(libraries) == 0:
             return None
         return QualifiedCommit(commit=commit, libraries=libraries)
+
+    @staticmethod
+    def __is_qualified_build_change(commit: Commit, build_file_path: str) -> bool:
+        """
+        Checks if the given commit containing a BUILD.bazel change is a
+        qualified commit.
+
+        The commit is a qualified commit if the
+        :class:`library_generation.model.gapic_inputs.GapicInputs` objects
+        parsed from the commit and its parent are different, since there are
+        changes in fields that used in library generation.
+
+        :param commit: a GitHub commit object.
+        :param build_file_path: the path of the BUILD.bazel
+        :return: True if the commit is a qualified commit; False otherwise.
+        """
+        versioned_proto_path = find_versioned_proto_path(build_file_path)
+        build = str((commit.tree / build_file_path).data_stream.read())
+        parent_commit = commit.parents[0]
```

Comment:
I know we are getting it from `commit.parents`, but conceptually I guess this is the last/previous commit?

Reply:
Maybe.

However, there might be multiple commits changed a `BUILD.bazel` between baseline and current commit.

Reply:
> there might be multiple commits changed a BUILD.bazel between baseline and current commit

Can you elaborate on this with an example?

Reply:
> Can you elaborate on this with an example?

I don't have a concrete example but now I understand your original question.

Yes, the parent commit contains the last commit touching this file.